plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/explorer/OpenShiftExplorerLabelProvider.java

```diff
@@ -109,14 +113,25 @@ public StyledString getStyledText(Object element) {
 				break;
 			}
 		}
-		if (element instanceof ResourceGrouping) {
-			return getStyledText((ResourceGrouping) element);
-		} else if (element instanceof Connection) {
+		 if (element instanceof Connection) {
 			return getStyledText((Connection) element);
 		}
+		if (element instanceof Deployment) {
+			Deployment d = (Deployment) element;
+			return style(d.getService().getName(), formatRoute(d.getRoutes()));
+		}
 		return super.getStyledText(element);
 	}
 
+	private String formatRoute(Collection<IRoute> routes) {
+		if(routes.size() > 0) {
+			IRoute route = (IRoute)routes.toArray()[0];
```

Comment:
Yes, I agree that this is a micro-optimization which in this concrete case where 'routes' list may not be be large, does not deserve much attention.
But, it still may be a good occasion to assert that list.iterator().next() is _always_ faster than list.toArray()[0], even for the size 1. And if it comes to a habit to convert to list when one element is needed, the habit may play a bad joke in a performance bottleneck case, as in some cases times do not just add but multiply.